tests/test_modules/test_dataset_analysis/test_syntax_tagging.py

```diff
@@ -2,16 +2,18 @@
 # This source code is licensed under the Apache 2.0 license found in the LICENSE file
 # in the root directory of this source tree.
 from azimuth.modules.dataset_analysis.syntax_tagging import SyntaxTaggingModule
-from azimuth.types import DatasetSplitName
+from azimuth.types import DatasetColumn, DatasetSplitName
 from azimuth.types.tag import SmartTag
 
 
-def test_syntax_tagging(tiny_text_config):
-    mod = SyntaxTaggingModule(
-        DatasetSplitName.eval,
-        tiny_text_config,
-    )
+def verify_syntax_results(json_output, smart_tags_per_idx, token_count_per_idx):
+    for idx, res in enumerate(json_output):
+        assert all(v for k, v in res.tags.items() if k in smart_tags_per_idx[idx])
+        assert not any(v for k, v in res.tags.items() if k not in smart_tags_per_idx[idx])
+        assert res.adds[DatasetColumn.word_count] == token_count_per_idx[idx]
```

Comment:
In fact, instead of `enumerate()` and using `idx`, you could `zip()`.
```suggestion
    for res, smart_tags, token_count in zip(json_output, smart_tags_per_idx, token_count_per_idx):
        assert list(res.tags.values()) == [k in smart_tags for k in res.tags.keys()]
        assert res.adds[DatasetColumn.word_count] == token_count
```
It could even be `zip_longest()` to be more robust if the response is shorter than the expected lists.

Reply:
I thought about the zip, but I didn't see any reason why it would be preferable. Could you please tell me more about why you prefer it?
For `zip_longest()`, I think that it should break if a list is shorter.

Reply:
Exactly. With `zip_longest()`, if any one list is too short, it will be "padded" with `None`, ensuring the `assert` still happens. Right now, if `json_output` is an empty list, nothing will get tested. That is why I would suggest the method with `zip_longest`.

Reply:
But we do assert that `json_output` has the right length just before this function so that it is never empty, right? I can make the change, I'm just a bit puzzled still. Reading this code, I wouldn't understand why we use 'zip_longest' given that the 3 lists should be equal for the test/function to make sense.

Reply:
Correct. So it's more for developer safety, in the event the test changes later. Right now, `verify_syntax_results()` is a reusable function, but nothing forces the developer to make sure the 3 lengths are equal before calling it. If someone adds a new test and reuses `verify_syntax_results()`, one could think it will check all that based on `verify_syntax_results()`'s name and forget to check the lengths before calling. But I agree it's not necessary. It should never pose a problem as long as we don't change the test.

Reply:
Ok I get it now, thanks!!